### PR TITLE
Restructure project setup

### LINKED ISSUE #2: Project could benefit from an enhanced structure
# Project Structure

According to the  "2020-TaskDocument", teams are required to build on top of the provided **Template Package** during development. However, the given project directory layout and configuration contradict some of the good practices laid out in  Maven's official documentation. For instance, the location of the application sources at ```src/main/java``` and tests at ```src/test/java```.

Reference:
https://maven.apache.org/guides/introduction/introduction-to-the-standard-directory-layout.html

Consequently, the team has decided to apply structural changes to the project in order to improve the separation of concerns, comply with Maven's good practices, and enhance the development experience.

Such changes include:
- Proper configuration of the project to meet the requirements stated in the "2020-TaskDocument"
- Revamp of the [directory layout](https://trello.com/c/ZzaPqo3o)
- Creation of empty template classes to kickstart the implementation of our [architecture](https://trello.com/c/bujzgVIg)

### PR DESCRIPTION
**Pull request by BelfDev**
# Project Setup

- [x] Revamp the project directory layout to comply with Maven [best practices](https://maven.apache.org/guides/introduction/introduction-to-the-standard-directory-layout.html)
- [x] Adjust project configuration according to "2020-TaskDocument" requirements
     - Change the project name: MScIT2020Project → TopTrumps
     - Remove "mysql-connector-java" dependency
     - Add other configurations (e.g., build, test resource, etc.)
- [x] Create empty template classes to guide through the development

Fix #2

### CODE REVIEW

File: src/main/java/com/toptrumps/cli/TopTrumpsCLIApp.java [hunk 1]
```diff
@@ -1,18 +1,15 @@
...
-
-package commandline;
-import core.*;
-
+import com.toptrumps.core.Deck;
 
 /**
  * Top Trumps command line application
  */
-public class TopTrumpsCLIApplication {
```
Comment (hughwin) [inline src/main/java/com/toptrumps/cli/TopTrumpsCLIApp.java:10]: Are you sure renaming this is a good idea if they use automated tests?

**Review by hughwin — COMMENTED** [review]
Looks good!

File: src/main/java/com/toptrumps/cli/TopTrumpsCLIApp.java (hunk 1 above)
Comment (BelfDev) [reply src/main/java/com/toptrumps/cli/TopTrumpsCLIApp.java:10]: We can double-check with the professor in the next lecture. But anyway, they were not crystal clear in the task document that we couldn't rename the classes.

> After submission your program will be tested, so it is particularly important that your program does not refer to the absolute location of the deck data file. When testing your program within Eclipse, store the deck data file in the current working directory.

**Review by nathanieldschneider — APPROVED** [review]
Greatest code I have ever seen in my life. The next Bill Gates. Truly. Anyway yeah, I too have concerns about changing names since it seems very arbitrary on things we do and do not have control over. But I trust your judgement so go for it.

**Comment by rjs68** [comment]
I’ve only had a quick look through but it looks good to me!

**Review by hughwin — APPROVED** [review]

**Review by rjs68 — APPROVED** [review]

**Review by Arden488 — APPROVED** [review]

File: src/main/java/com/toptrumps/cli/TopTrumpsCLIApp.java (hunk 1 above)
Comment (BelfDev) [reply src/main/java/com/toptrumps/cli/TopTrumpsCLIApp.java:10]: @hughwin I spoke with the lecturer and he said we're fine! We just need to provide a mapping table on our report :)

**Pull request merged**